src/main/kotlin/com/nikitanazarov/coroutinestacks/ui/CoroutineFramesList.kt

```diff
@@ -0,0 +1,128 @@
+package com.nikitanazarov.coroutinestacks.ui
+
+import com.intellij.debugger.engine.SuspendContextImpl
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.ui.JBColor
+import com.intellij.ui.JBColor.BLACK
+import com.intellij.ui.JBColor.BLUE
+import com.intellij.ui.components.JBList
+import com.intellij.util.ui.JBUI
+import com.intellij.xdebugger.frame.XExecutionStack
+import com.intellij.xdebugger.frame.XStackFrame
+import com.nikitanazarov.coroutinestacks.CoroutineTrace
+import org.jetbrains.kotlin.analysis.decompiler.stub.file.ClsClassFinder
+import org.jetbrains.kotlin.idea.debugger.coroutine.view.SimpleColoredTextIconPresentationRenderer
+import java.awt.*
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
+import javax.swing.BorderFactory
+import javax.swing.DefaultListCellRenderer
+import javax.swing.JComponent
+import javax.swing.JList
+import javax.swing.border.Border
+import javax.swing.border.LineBorder
+
+class CoroutineFramesList(
+    suspendContext: SuspendContextImpl,
+    trace: CoroutineTrace
+) : JBList<String>() {
+    companion object {
+        private val itemBorder = BorderFactory.createMatteBorder(0, 0, 1, 0, JBColor.GRAY)
+        private val leftPaddingBorder: Border = JBUI.Borders.emptyLeft(3)
+        private val compoundBorder = BorderFactory.createCompoundBorder(itemBorder, leftPaddingBorder)
+        private const val cornerRadius = 10
+        private const val borderThickness = 1
+    }
+
+    init {
+        val debugProcess = suspendContext.debugProcess
+
+        val data = mutableListOf<String>()
+        data.add(trace.header)
+        val renderer = SimpleColoredTextIconPresentationRenderer()
+        data.addAll(trace.stackFrameItems.map {
+            renderer.render(it!!.location).simpleString()
```

Comment:
Let's just add an empty string if we accidentally get a null. Throwing exceptions from a constructor is a very bad pattern!